src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -407,7 +407,7 @@ def get_system_mac(namespace=None):
         hw_mac_entry_cmds = [mac_address_cmd]
 
     for get_mac_cmd in hw_mac_entry_cmds:
-        proc = subprocess.Popen(get_mac_cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        proc = subprocess.Popen(get_mac_cmd, shell=True, universal_newlines=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
Thanks for the detailed investigation! So the best method should be:
1. No extra argument for python 2
2. Add `text=True` for python 3

I see you tried hard to maintain same statement for different version, actually it is not that important. Instead, I suggest you use a `kwargs` variables, add options based on python version, and pass it to `Popen` constructor.

Reply:
I don't think it's work this much effort and refactoring, as we will be removing support for Python 2 soon. I think we should start with this solution as it's backward-compatible, then change `universal_newlines=True` to `text=True` once we support Python 3 only.